pkg/scheduler/framework/interface.go

```diff
@@ -160,6 +162,27 @@ func (s *Status) AsError() error {
 	return errors.New(s.Message())
 }
 
+// StatusEqual returns whether two Statuses are equal to each other.
+func StatusEqual(x, y *Status) bool {
+	if x == nil || y == nil {
+		return x.IsSuccess() && y.IsSuccess()
+	}
+	if x.code != y.code {
+		return false
+	}
+	if x.err == nil && y.err == nil {
+		if !cmp.Equal(x.reasons, y.reasons) {
+			return false
+		}
+	}
+	if x.err != nil && y.err != nil {
+		if cmp.Equal(x.err.Error(), y.err.Error()) {
```

Comment:
this is comparing strings, don't do that. The comparison after this block should be enough.